### DISCUSSION TITLE
複数記事をまとめた目次作成機能

### DISCUSSION THREAD

**Discussion by hanaseleb** [increments/qiita-discussions#426] [要望・アイデア・不具合]
数本の投稿に分かれているシリーズ記事を投稿する際に、冒頭にシリーズ記事へのリンクを置きたいときが多くあります。

https://qiita.com/akihiro_suto/items/aaf2e4c98cf88833dc63

参考までに、私が最近投稿した記事でいうと、上記のように、手動でリンクを切り貼りして作成しています。

手動運用のため、記事タイトルを変更したときや、新規記事を投稿したときに過去記事全て変更しなければいけなかったり、運用が大変です。

ただ、読者的には、シリーズ記事なのであれば、リンクは貼ってあったほうがいいでしょう。

ということで、自分が書いた記事をポチポチっと選択するだけで記事一覧のMarkdownが吐き出される機能をつけてもらえると、関連記事リンクのメンテナンスが楽になり助かるなぁと思っています。

**Comment by tanaka-asami415** [comment]
@hanaseleb 
ご意見ありがとうございます🙏ご不便をおかけして、申し訳ありません。
直近の開発優先度などから着手はできておりませんが
同様のご意見もいただいており、運営としてもぜひやりたいと思っております！✨

**Reply by tfukumori** [reply]
もし実装される場合は、2階層程度で良いですので階層まであると嬉しいです
（実装が遅れるようなら1階層で構いません！）

イメージ
https://qiita.com/tfukumori/items/65056a4d51b531cdd45c

**Reply by tanaka-asami415** [reply]
@tfukumori 
詳細なイメージもありがとうございます🙏
参考にさせていただきます！

**Comment by kimkim0814** [comment]
@hanaseleb
大変お待たせしました！

以前いただいたリクエストの内容とは異なるのですが、
今回ベータ版でリリースした「ストックのシェア機能」をご利用いただくことで、いただいたリクエストを叶えられるのではないかと思います。

シリーズ物の記事をストックリストとしてまとめ、そのURLを記事に記載いただくことで、以前より管理と紹介が楽になるのではないかと思います。
残念ながら、ベータ版ではまだリストの並べ替えはできないのですが、
これから正式リリースに向けて、改善を予定しています！

ぜひ新機能をご活用いただき、気づいたことやご意見をフィードバックいただけると嬉しいです✨
[詳細はこちらをご覧ください。](https://blog.qiita.com/stock-share-beta/)

リクエストいただきありがとうございました！

**Comment by kimkim0814** [comment]
@hanaseleb
大変お待たせしました。

**「ストックの記事の並び替え機能」** も追加し、ストックのシェア機能の正式版として公開しました。

ストックリスト内で記事をお好みの順序に整理できるようになっています。シリーズ記事の管理として、ぜひご活用いただけると嬉しいです✨

詳しくは[こちら](https://blog.qiita.com/release-stock-share/)をご覧ください。

**Comment by scivola** [comment]
ストックのシェア機能そのものは歓迎です。ありがとうございます。
これによってシリーズ記事の扱いが改善されるのは確かですね。
ただ，もともと hanaseleb さんが提起された課題（記事内の目次作成）はなんら解決しないように思います。

私もシリーズ記事（間のナビゲーション）をどう書くかに悩んできました。
現時点の仕様に沿って何がベストか考えるため，試しに自分のシリーズ記事のストックリストを作って，各記事の冒頭にリンクを入れてみました。
[Ruby/Rust 連携 (8)　Rust 拡張 gem を magnus で作る #rubygems - Qiita](https://qiita.com/scivola/items/0247c0d518633961924e)

ストックリストがシリーズ記事のものであることと分かるよう，ストックリストの名前の頭に「シリーズ　」を付けてみました。（逆に言えば，こうしないと分かりづらい）
ストックリストのページの表示は，一つ一つの記事の高さがけっこうあるので，一覧性が悪く感じます。4、5 本ならまだいいですが，10 本くらいになるとどういうシリーズなのか頭に入ってこない感じです。
そういうこともあって，シリーズの各記事の冒頭に置いていた目次（全記事へのリンク）は削除せずに残しました。
そうすると，新しい記事を追加するたびに過去記事全部を編集し直す必要があります。
これはたいへん面倒ですし，目次しか触っていない古い記事の更新日が「今日」になるのもどうかと思います。

**Reply by kimkim0814** [reply]
@scivola
ストックのシェア機能をお試しいただき、そして詳細なフィードバックをいただき、ありがとうございます。

おっしゃる通り、今回ご提案した機能は元々のリクエストを直接解決するものではなく、あくまで代替案となります。

いただいたフィードバックを通じて、ストックリストページの一覧性の問題や、手動更新の手間といったご不便をおかけする点があることを、運営としても改めて認識いたしました。

すぐに改善をお約束できませんが、今後のサービス改善に活かしてまいります🙇

この度は貴重なご意見をいただき、本当にありがとうございました！
